app/src/main/java/com/github/steroidteam/todolist/ItemViewActivity.java

```diff
@@ -35,22 +37,55 @@ protected void onCreate(Bundle savedInstanceState) {
         ActionBar ab = getSupportActionBar();
         ab.setDisplayHomeAsUpEnabled(true);
 
-        String todoListTitle = "Example list";
-        setTitle(todoListTitle);
+        TodoList todoList = new TodoList("Example list");
+        setTitle(todoList.getTitle());
 
         // Pre-populate the database with a few sample tasks.
-        List<Task> tasks = new ArrayList<>();
-        tasks.add(new Task("Change passwords"));
-        tasks.add(new Task("Replace old server"));
-        tasks.add(new Task("Set up firewall"));
-        tasks.add(new Task("Fix router"));
-        tasks.add(new Task("Change passwords"));
-        tasks.add(new Task("Replace old server"));
-        tasks.add(new Task("Set up firewall"));
-
-        adapter = new TasksAdapter(this, tasks);
+        todoList.addTask(new Task("Change passwords"));
+        todoList.addTask(new Task("Replace old server"));
+        todoList.addTask(new Task("Set up firewall"));
+        todoList.addTask(new Task("Fix router"));
+        todoList.addTask(new Task("Change passwords"));
+        todoList.addTask(new Task("Replace old server"));
+        todoList.addTask(new Task("Set up firewall"));
+
+        //adapter = new TasksAdapter(this, tasks);
+        adapter = new todoAdapter(todoList);
         ListView listView = findViewById(R.id.activity_itemview_itemlist);
+        setListViewSettings(listView, todoList);
+    }
+
+    private void setListViewSettings(ListView listView, TodoList todoList) {
         listView.setAdapter(adapter);
+
+        listView.setLongClickable(true);
+        // The part for deleteTask
+        listView.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> adapterView, View view, int i, long l) {
+
+                AlertDialog.Builder builder = new AlertDialog.Builder(ItemViewActivity.this);
+
+                builder.setTitle("You are about to delete a task!")
+                        .setMessage("Are you sure ?")
+                        .setNegativeButton("No", new DialogInterface.OnClickListener() {
+                            @Override
```

Comment:
If there's nothing to be done, I'd say that we can remove the event listener entirely.